lib/bone.js

```diff
@@ -144,17 +163,57 @@ class Bone {
     }
     if (args.length > 1) {
       // execute validators
-      this.raw[name] = value;
-      this.rawUnset.delete(name);
+      this.setRaw(name, value);
+      this.getRawUnset().delete(name);
       return this;
     }
-    if (this.rawUnset.has(name)) throw new Error(`unset attribute "${name}"`);
-    const rawValue = this.raw[name];
+    if (this.getRawUnset().has(name)) throw new Error(`unset attribute "${name}"`);
+    const rawValue = this.getRaw(name);
     // make sure null is returned if value is undefined
     return rawValue == null ? null : rawValue;
 
   }
 
+  setRaw(key, value) {
+    this[raw][key] = value;
+  }
+
+  getRaw(key) {
+    if (key) return this[raw][key];
+    return this[raw];
+  }
+
+  getRawUnset() {
+    return this[rawUnset];
+  }
+
+  getRawSaved(key) {
+    if (key) return this[rawSaved][key];
+    return this[rawSaved];
+  }
+
+  setRawSaved(key, value) {
+    this[rawSaved][key] = value;
+  }
+
+  getRawPrevious(key) {
+    if (key) return this[rawPrevious][key];
+    return this[rawPrevious];
+  }
+
+  setRawPrevious(key, value) {
+    this[rawPrevious][key] = value;
+  }
+
+  isNewRecord() {
+    return this[isNewRecord];
+  }
+
+  setIsNewRecord(value) {
+    this[isNewRecord] = value;
+  }
```

Comment:
just a second thought, is turning `isNewRecord` into symbol really necessary?

Reply:
if not so it will be serialized while calling `toJSON` or `toObject`